client/components/common/ItemCard.tsx

```diff
@@ -33,38 +35,64 @@ const ItemCard: React.FC<IProps> = ({
   selectItemSlot,
   equipped,
   openSetModal,
-  isMobile,
+  shouldRedirect,
   nextSlotId,
 }) => {
   const mutate = useEquipItemMutation(item);
 
   const client = useApolloClient();
 
-  const onClick = React.useCallback(() => {
+  const router = useRouter();
+  const { query } = router;
+
+  const { t } = useTranslation('common');
+
+  const onClick = React.useCallback(async () => {
     if (itemSlotId) {
       const slots = client.readQuery<itemSlots>({ query: ItemSlotsQuery });
       let nextSlot = null;
       if (nextSlotId && slots) {
         nextSlot = slots.itemSlots.find(slot => slot.id === nextSlotId) || null;
       }
       selectItemSlot && selectItemSlot(nextSlot);
-      if (isMobile && customSetId) {
-        Router.push(
-          { pathname: '/index', query: { customSetId } },
-          customSetId ? `/build/${customSetId}` : '/',
-        );
+      if (shouldRedirect && customSetId) {
+        if (nextSlot) {

```

Comment:
If there is still another empty slot of the same type, stay on the same page and show a notification to the user to show that the item has been equipped.